include/vcpkg/base/messages.h

```diff
@@ -128,15 +128,33 @@ VCPKG_FORMAT_AS(vcpkg::LocalizedString, vcpkg::StringView);
 
 namespace vcpkg::msg
 {
+    namespace detail
+    {
+        template<class... Tags, class... Types, std::size_t... S>
+        LocalizedString format_impl(std::index_sequence<S...>, MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
+        {
+            auto format_args = std::make_tuple(args.arg()...);
+            (void)format_args;
+            return detail::format_message_by_index(m.index, fmt::make_format_args(std::get<S>(format_args)...));
+        }
+        template<class... Tags, class... Types, std::size_t... S>
+        void format_to_impl(std::index_sequence<S...>, LocalizedString& s, MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
+        {
+            auto format_args = std::make_tuple(args.arg()...);
+            (void)format_args;
+            return detail::format_message_by_index_to(s, m.index, fmt::make_format_args(std::get<S>(format_args)...));
+        }
+    }
+    
     template<class... Tags, class... Types>
     LocalizedString format(MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
     {
-        return detail::format_message_by_index(m.index, fmt::make_format_args(args.arg()...));
+        return detail::format_impl(std::index_sequence_for<Types...>{}, m, args...);
     }
     template<class... Tags, class... Types>
     void format_to(LocalizedString& s, MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
     {
-        return detail::format_message_by_index_to(s, m.index, fmt::make_format_args(args.arg()...));
+        return detail::format_to_impl(std::index_sequence_for<Types...>{}, s, m, args...);
     }
```

Comment:
I don't see any reason to involve tuple here. How about:

```
    namespace detail
    {
        template<class... FmtArgs>
        LocalizedString format_impl(size_t index, FmtArgs&&... args)
        {
            // no forward to intentionally make an lvalue here
            return detail::format_message_by_index(index, fmt::make_format_args(args...));
        }

        template<class... FmtArgs>
        void format_to_impl(LocalizedString& s, size_t index, FmtArgs&&... args)
        {
            // no forward to intentionally make an lvalue here
            return detail::format_message_by_index_to(s, index, fmt::make_format_args(args...));
        }
    }

    template<class... Tags, class... Types>
    LocalizedString format(MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
    {
        return detail::format_impl(m.index, args.arg()...);
    }
    template<class... Tags, class... Types>
    void format_to(LocalizedString& s, MessageT<Tags...> m, TagArg<identity_t<Tags>, Types>... args)
    {
        return detail::format_to_impl(s, m.index, args.arg()...);
    }
```

Reply:
Oh! Yeah makes sense. Let me change that!